src/app.rs

```diff
@@ -155,9 +189,10 @@ impl<'repo> App<'repo> {
             NavigationAction::PreviousSelection => page.previous_selection(),
             NavigationAction::Invalid => {}
         }
+        Ok(())
     }
 
-    pub fn select(&mut self) {
+    pub fn select(&mut self) -> Result<(), GitBrowserError> {
```

Comment:
What do you think about exiting early and reducing the nesting? Maybe something similar to,

```suggestion
    pub fn select(&mut self) -> Result<(), GitBrowserError> {
        if self.blob_pager.is_some() {
            return Ok(())
        }

        let page: Box<&dyn Navigable> = if let Some(p) = self.tree_pages.last() {
            Box::new(p)
        } else {
            Box::new(&self.refs_page)
        };

        let (object, name) = match page.select() {
            Some(selection) => selection,
            None => return Ok(()),
        };

        return match object.kind() {
            Some(ObjectType::Blob) => {
                let pager = BlobPager::from_object(self.repo, object, page.selected_item())?;
                self.blob_pager = Some(pager);
                Ok(())
            }
            Some(ObjectType::Tree) => {
                self.tree_pages.push(TreePage::new(self.repo, object, name));
                Ok(())
            }
            Some(ObjectType::Commit) => {
                let commit = object.peel_to_commit().expect("Unable to peel commit");
                self.commit = Some(commit);
                self.tree_pages.push(TreePage::new(self.repo, object, name));
                Ok(())
            },
            _ => Ok(())
        }
    }
```